internal/librarian/commit_version_analyzer_test.go

```diff
@@ -301,6 +337,85 @@ func TestGetConventionalCommitsSinceLastRelease(t *testing.T) {
 	}
 }
 
+func TestGetConventionalCommitsSinceLastGeneration(t *testing.T) {
+	t.Parallel()
+	for _, test := range []struct {
+		name          string
+		repo          gitrepo.Repository
+		library       *config.LibraryState
+		want          []*conventionalcommits.ConventionalCommit
+		wantErr       bool
+		wantErrPhrase string
+	}{
+		{
+			name: "found_matching_file_changes_for_foo",
+			library: &config.LibraryState{
```

Comment:
Could you add a test case to verify source roots have no impact to the result.

Reply:
Will do